### PR TITLE
[WIP] Make checking optional for dependent encoding profiles

### LINKED ISSUE #160: Disable checking in encoding profile
It should be possible to disable checking for certain encoding profiles. On many events the "subformats" are released without checking.

### PR DESCRIPTION
**Pull request by zuntrax**
Closes #160 
Resubmitted from #162

### CODE REVIEW

File: src/Application/Migrations/15_function_ticket_state.sql [hunk 1]
```diff
@@ -2,7 +2,7 @@ BEGIN;
 
 SET ROLE TO postgres;
 
-CREATE OR REPLACE FUNCTION ticket_state_next(param_project_id bigint, param_ticket_type enum_ticket_type, param_ticket_state enum_ticket_state)
+CREATE OR REPLACE FUNCTION ticket_state_next(param_ticket_id bigint)
```
Comment (zuntrax) [inline src/Application/Migrations/15_function_ticket_state.sql]: @jjeising @pegro 
[Continuing discussion](https://github.com/crs-tools/tracker/pull/162#discussion_r183407717)
This doesn't work, `Ticket::expandRecording` calls `Ticket::queryPreviousState` with `state='preparing'`, differing from the current state of the ticket, which would usually be `cutting`.
I'd suggest using an additional parameter `skip_dependent DEFAULT FALSE` and adding a new function (maybe with a trigger as suggested) `ticket_is_master` to pass that in from outside and keep this function completely ticket agnostic.

Comment (pegro) [reply src/Application/Migrations/15_function_ticket_state.sql]: As I said before, if determining the next state depends on more as just the ticket type and state, I'm fine with changing all functions to just requiring a ticket_id and migrate all users accordingly.

File: src/Application/Model/TicketState.php [hunk 2]
```diff
@@ -31,6 +33,11 @@ public static function getStateByAction($action) {
 			
 			return self::$_actions[$action];
 		}
+
+		public static function isSkippable($state) {
+			return ($state === 'postencoded' ||
+				$state === 'checking');
```
[the remark this answers is not in the document]
Comment (jjeising) [reply src/Application/Model/TicketState.php]: I think this should be moved to `tbl_state`?

File: src/Application/Migrations/15_function_ticket_state.sql (hunk 1 above)
Comment (jjeising) [reply src/Application/Migrations/15_function_ticket_state.sql]: >  I'm fine with changing all functions to just requiring a ticket_id

The new issue here is: ticket_state_next is also used with a state different from the current state of the ticket (with `param_ticket_id`).

Comment (pegro) [reply src/Application/Migrations/15_function_ticket_state.sql]: If you want to query for all possible ticket states of the current ticket, I would then add a second optional parameter param_state as a state of reference which defaults to the current ticket state.

So you could call ticket_state_next(param_ticket_id) to get the next state in relation to the current state. And you could call ticket_state_next(param_ticket_id, param_ticket_state) to query for the next state the ticket would get advanced to, **if** the ticket would be in state _param_ticket_state_.

At least that's what I'd suggest.

File: src/Application/Model/TicketState.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (pegro) [reply src/Application/Model/TicketState.php]: I guess you mean ```tbl_ticket_state```, but yes.

Another question: should this setting only be visible if any profiles with a dependency are assigned to the project?

File: src/Application/Migrations/15_function_ticket_state.sql (hunk 1 above)
Comment (jjeising) [reply src/Application/Migrations/15_function_ticket_state.sql]: @zuntrax That's what we discussed previously, can you implement that?

File: src/Application/Model/TicketState.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (jjeising) [reply src/Application/Model/TicketState.php]: > Another question: should this setting only be visible if any profiles with a dependency are assigned to the project?

I'm not sure if it is worth introducing additional complexity in the view. It's only two checkboxes at the moment, which have a default value.

File: src/Application/Migrations/15_function_ticket_state.sql [hunk 3]
```diff
@@ -111,7 +164,7 @@ BEGIN
   ret := (SELECT ticket_state_initial(param_project_id, param_ticket_type));
 
   WHILE ret IS NOT NULL LOOP
-    SELECT * INTO next_state FROM ticket_state_next(param_project_id, param_ticket_type, ret);
+    SELECT * INTO next_state FROM ticket_state_next(param_ticket_id);
```
Comment (a-tze) [inline src/Application/Migrations/15_function_ticket_state.sql]: Note: eliminating ret from parameters will probably produce endless loops.

**Comment by jjeising** [comment]
Superseded by #217.

**Pull request closed without merging**